Book-A-Desk.Api/EmailNotification.fs

```diff
@@ -0,0 +1,46 @@
+namespace Book_A_Desk.Api
+
+open System
+open System.Net.Mail
+open Book_A_Desk.Api.Models
+open Book_A_Desk.Domain.QueriesHandler
+open Book_A_Desk.Domain.Office.Domain
+
+type EmailNotification =
+    {
+        SendEmailNotification: Booking -> unit
+    }
+
+module EmailNotification =
+    let initialize getEmailServiceConfiguration getOffices =         
```

Comment:
Why is getEmailServiceConfiguration a function and not simply the configuration itself? Do we have to fetch the configuration at runtime?

Reply:
It is because it needs to be executed deferred as the container where the configuration is registered is not fully set up when the EmailNotification module gets initialized. That's why we grab it from the container when it is needed per function.